File: strings/countingWordsWithGivenPrefix.java
package strings;

public class countingWordsWithGivenPrefix {
    public static void main(String[] args) {

        SolutionCWWGP solution = new SolutionCWWGP();
        System.out.println(solution.prefixCount(new String[]{"leetcode","win","loops","success"}, "code"));
        System.out.println(solution.prefixCount(new String[]{"pay","attention","practice","attend"}, "at"));

    }
}

class SolutionCWWGP {

    // Submitted by Jiganesh 

    // TC : O(N * M)
    // SC : O(N)

    // Runtime: 1 ms, faster than 83.33% of Java online submissions for Counting Words With a Given Prefix.
    // Memory Usage: 43.7 MB, less than 50.00% of Java online submissions for Counting Words With a Given Prefix.
    
    public int prefixCount(String[] words, String pref) {
        
        int count = 0;
        
        int lengthOfPref = pref.length();
        
        for (String word: words){

            // check if length of prefix is less or equal to word else there is not point in checking
            // extract the substring pref from word and check if it is equal to pref
            if (lengthOfPref<= word.length() && word.substring(0, lengthOfPref).equals(pref)){
                count++;
            }
        }
        return count;  
    }
}

File: strings/countingWordsWithGivenPrefix.py
class Solution(object):
    
    # Submitted by Jiganesh
    
    # TC: O(N*M)
    # SC : O(N)
    
    # Runtime: 33 ms, faster than 50.00% of Python online submissions for Counting Words With a Given Prefix.
    # Memory Usage: 13.5 MB, less than 100.00% of Python online submissions for Counting Words With a Given Prefix.
    
    # one liner Solution
    def prefixCount(self, words, pref):
        """
        :type words: List[str]
        :type pref: str
        :rtype: int
        """
        return len( [ i for i in words if i.startswith(pref)])
    
print(Solution().prefixCount(["pay","attention","practice","attend"],"at"))
print(Solution().prefixCount(["pay","lecture","pen","leetcode","codearena"],"code"))